frontend/tests/acceptance/pages/timeLogPage.js

```diff
@@ -9,13 +10,171 @@ module.exports = {
   elements: {
     timelog_title: '//h6[text()="Time Logs"]',
     empty_message: '//h5[text()="No timelogs added yet"]',
-    openaddtimelog_dialog: '//div[#aria-labelledby="Add time log"]',
+    openaddtimelog_dialog: '//button/span[contains(text(),"Add")]',
     openedittimelog_button: '//span[@aria-label="Edit time log"]',
-    addtimelog_button: '//button/span[contains(text(),"Add")]',
-    updatetimelog_button: '//button/span[contains(text(),"Update")]',
+    addtimelog_button: '//div[contains(@class,"MuiDialogContent-root")]//button/span[contains(text(),"Add")]',
+    updatetimelog_button: '//div[contains(@class,"MuiDialogContent-root")]//button/span[contains(text(),"Update")]',
     deletetimelog_button: '//span[@aria-label="Delete time log"]',
-    addtocalender_button: '//span[@label="Add to calender"]',
+    addtocalendar_button: '//span[@aria-label="Add to calender"]',
     select_options: '//div[contains(@class,"MuiPopover-paper")]//li',
-    timelog_container: '//div[contains(@class, "makeStyles-timeLogItem")]',
+    timelog_container: '//h6[text()="Time Logs"]/parent::div',
+    timelog_dialog: '//div[@id="add-title"]',
+    getTimeLogContext(type) {
+      return `//p[text()="${type.toUpperCase()}"]/parent::div`;
+    },
+  },
+  datePicker: {
+    datePicker_container: '//div[contains(@class,"MuiPickersBasePicker-container")]',
+    open_year: '//div[1]/button[contains(@class,"MuiPickersToolbarButton-toolbarBtn")]//h6',
+    open_day_month: '//button[contains(@class,"MuiPickersToolbarButton-toolbarBtn")][1]//h4',
+    open_hour: '//button[contains(@class,"MuiPickersToolbarButton-toolbarBtn")][1]//span/h3',
+    open_minute: '//button[contains(@class,"MuiPickersToolbarButton-toolbarBtn")][2]//span/h3',
+    month_container: '//div[contains(@class,"MuiPickersSlideTransition-transitionContainer")]/p',
+    next_month: '//div[contains(@class,"MuiPickersCalendarHeader-switchHeader")]/button[2]//*[name()="svg"]',
+    previous_month: '//div[contains(@class,"MuiPickersCalendarHeader-switchHeader")]/button[1]//*[name()="svg"]',
+    getYear(year) {
+      return `//div[contains(@class,"MuiPickersYearSelection-container")]/div[text()="${year}"]`;
+    },
```

Comment:
use variables rather functions!
```suggestion
    year_selector: '//div[contains(@class,"MuiPickersYearSelection-container")]/div[text()="%s"]'
```
then use anywhere as,
```js
util.format(year_selector, year)
```

Reply:
what might be the benefits. as I have to import a module